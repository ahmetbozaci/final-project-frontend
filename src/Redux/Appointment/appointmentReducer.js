const CREATE_APPOINTMENT = 'final-capstone-project/users/CREATE_APPOINTMENT';

const initialState = [];

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case CREATE_APPOINTMENT:
      return action.payload;
    default:
      return state;
  }
};

export const createAppointmentAction = (apiState) => ({
  type: CREATE_APPOINTMENT,
  payload: apiState,
});

export default reducer;
