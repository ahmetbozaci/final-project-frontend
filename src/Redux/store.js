import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from './rootReducer';
import fetchDoctors from './doctor/doctorActions';

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk, logger)),
);
store.dispatch(fetchDoctors());

export default store;
