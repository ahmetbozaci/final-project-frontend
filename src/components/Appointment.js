import React from 'react';
import { Form, Button } from 'react-bootstrap';

const Appointment = () => (
  <Form>
    <br />
    <br />
    {/* Disable weekends */}
    <h1>Appointment</h1>
    <Form.Group className="mb-3" controlId="formBasicEmail">
      <Form.Control type="date" min={Date.now()} required />
    </Form.Group>
    {/* Disable minutes */}
    <Form.Group className="mb-3" controlId="formBasicEmail">
      <Form.Control type="time" step="" min="09:00" max="17:00" required />
      <Form.Text className="text-muted">
        Choose time 09:00 AM to 05:00 PM
      </Form.Text>
    </Form.Group>
    {/* Doctor name will come from state */}
    <Form.Group className="mb-3" controlId="formBasicEmail">
      <Form.Control type="text" disabled value="Doctor Name" />
    </Form.Group>
    <Button variant="primary" type="submit">
      Create Appointment
    </Button>
  </Form>
);

export default Appointment;
