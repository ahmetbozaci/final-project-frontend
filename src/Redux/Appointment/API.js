import axios from 'axios';
import { createAppointmentSuccess } from './appointmentReducer';

const baseURL = `${process.env.REACT_APP_DOCTOR_APPOINTMENT_API_URL}/appointments`;

const createAppointment = (appointment) => async (dispatch) => {
  const newAppointment = {
    date: appointment.date,
    time: appointment.time,
    user_id: appointment.userID,
    doctor_id: appointment.doctorID,
  };

  const response = await axios.post(baseURL, newAppointment);
  const { data } = response;
  dispatch(createAppointmentSuccess(data));
};

export default createAppointment;
