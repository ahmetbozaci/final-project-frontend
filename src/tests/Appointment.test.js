import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import Appointment from '../components/Appointment';

it('Appointment renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><Router><Appointment /></Router></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
