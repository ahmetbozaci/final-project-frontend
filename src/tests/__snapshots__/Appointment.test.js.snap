// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointment renders correctly 1`] = `
<form
  className="appointmentMargin"
  onSubmit={[Function]}
>
  <br />
  <br />
  <h2
    className="pb-3"
  >
    Fill the form to create your appointment
  </h2>
  <div />
  <div
    className="mb-3"
  >
    <input
      className="form-control"
      id="formBasicEmail"
      name="date"
      onChange={[Function]}
      required={true}
      type="date"
      value=""
    />
  </div>
  <div
    className="mb-3"
  >
    <input
      className="form-control"
      id="formBasicEmail"
      name="time"
      onChange={[Function]}
      required={true}
      type="time"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Create Appointment
  </button>
</form>
`;
